ambry-tools/src/integration-test/java/com.github.ambry/clustermap/HelixBootstrapUpgradeToolTest.java

```diff
@@ -293,25 +345,29 @@ private void uploadClusterConfigsAndVerify() throws Exception {
     Utils.writeJsonObjectToFile(testHardwareLayout.getHardwareLayout().toJSONObject(), hardwareLayoutPath);
     Utils.writeJsonObjectToFile(testPartitionLayout.getPartitionLayout().toJSONObject(), partitionLayoutPath);
     HelixBootstrapUpgradeUtil.uploadClusterConfigs(hardwareLayoutPath, partitionLayoutPath, zkLayoutPath,
-        CLUSTER_NAME_PREFIX, DEFAULT_MAX_PARTITIONS_PER_RESOURCE, new HelixAdminFactory());
+        CLUSTER_NAME_PREFIX, dcStr, DEFAULT_MAX_PARTITIONS_PER_RESOURCE, new HelixAdminFactory());
     // Check writable partitions in each datacenter
     for (ZkInfo zkInfo : dcsToZkInfo.values()) {
       HelixPropertyStore<ZNRecord> propertyStore =
           CommonUtils.createHelixPropertyStore("localhost:" + zkInfo.getPort(), propertyStoreConfig,
               Collections.singletonList(propertyStoreConfig.rootPath));
       String getPath = ClusterMapUtils.PROPERTYSTORE_ZNODE_PATH;
       ZNRecord zNRecord = propertyStore.get(getPath, null, AccessOption.PERSISTENT);
-      assertNotNull(zNRecord);
-      Map<String, Map<String, String>> overridePartition = zNRecord.getMapFields();
-      Set<String> writableInDC = new HashSet<>();
-      for (Map.Entry<String, Map<String, String>> entry : overridePartition.entrySet()) {
-        if (entry.getValue().get(ClusterMapUtils.PARTITION_STATE).equals(ClusterMapUtils.READ_WRITE_STR)) {
-          writableInDC.add(entry.getKey());
+      if (!activeDcSet.contains(zkInfo.getDcName())) {
+        assertNull(zNRecord);
```

Comment:
In my humble opinion, the testing here need to be improved, it's better to make record have previous state and compare zNRecord from not specified DC to this previous state. An alternative way is to deploy to all DCs at the end of this method and get the previous record at the very beginning of the method.

Reply:
You are right in that there are many ways this can be improved. 
Here we are checking that the node never gets created if that dc is not part of the active set. I guess your suggestion is to see that if it is already created, then it does not get updated. Though, I do feel that it is an overkill, since it does not exercise any different code - basically, the constructor only initializes connections to active dcs. The same code is getting exercised in either case. Note that we are not interested in checking if other dcs get affected if admin connections are not made to them.

Let me know what you think, we can have an offline conversation if you wish.